nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1463,6 +1463,9 @@ public DocumentModel saveDocument(DocumentModel docModel) {
             doc = getSession().move(doc, doc.getParent(), name);
         }
 
+        // recompute the dirty state
+        dirty = docModel.isDirty();
+
```

Comment:
right, it's put on the map for the event listener (used by the DublinCoreListener) but it could be used in other places later.